tap_salesforce/__init__.py

```diff
@@ -573,7 +582,40 @@ def main_impl():
 
             state = build_state(args.state, catalog)
             do_sync(sf, catalog, state)
+    except SymonException as e:
```

Comment:
Including SymonException changes from https://github.com/symon-ai/tap-salesforce/pull/25, since it is needed for extra error handling of invalid filter, but that PR hasn't been merged.
@jayma91